lib/build_runner.zig

```diff
@@ -731,10 +745,22 @@ fn printTreeStep(
     }
 }
 
-fn checkForDependencyLoop(
+/// Traverse the dependency graph depth-first and make it undirected by having
+/// steps know their dependants (they only know dependencies at start).
+/// Along the way, check that there is no dependency loop, and record the steps
+/// in traversal order in `step_stack`.
+/// Each step has its dependencies traversed in random order, this accomplishes
+/// two things:
+/// - `step_stack` will be in randomized-depth-first order, so the build runner
+///   spawns steps in a random (but optimized) order
+/// - each step's `dependants` list is also filled in a random order, so that
+///   when it finishes executing in `workerMakeOneStep`, it spawns next steps
+///   to run in random order
+fn buildAndCheckGraphForDependencyLoop(
```

Comment:
why was this renamed to "buildAnd..." ? It's not building yet.

Reply:
Maybe I'm missing the right vocabulary, but it is building and checking the graph (adding edges between nodes) and also creating the output stack, not _only_ checking for loops.

Reply:
Could maybe named `construct` to differentiate from "building" the project (by calling steps' `make` functions)? (If that's the ambiguity to avoid.)